pkg/operator/hive/hive.go

```diff
@@ -173,6 +174,30 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h *resource.Helpe
 
 	}
 
+	// Remove old validatingwebhookconfigurations
+	removeValidatingWebhooks := []string{
+		"clusterdeployments.admission.hive.openshift.io",
+		"clusterimagesets.admission.hive.openshift.io",
+		"dnszones.admission.hive.openshift.io",
+		"selectorsyncsets.admission.hive.openshift.io",
+		"syncsets.admission.hive.openshift.io",
+	}
+	for _, webhookName := range removeValidatingWebhooks {
+		webhookConfig := &admissionregistrationv1beta1.ValidatingWebhookConfiguration{}
+		err := r.Get(context.Background(), types.NamespacedName{Name: webhookName}, webhookConfig)
+		if err != nil {
+			hLog.Infof("Validating webhook does not exist: %v")
+		} else {
+			err = r.Delete(context.Background(), webhookConfig)
+			if err != nil {
+				hLog.WithError(err).WithField("ValidatingWebhookConfiguration", webhookConfig).Error(
+					"error deleting outdated ValidatingWebhookConfiguration")
+				return err
+			}
+			hLog.WithField("ValidatingWebhookConfiguration", webhookName).Info("deleted outdated ValidatingWebhookConfiguration")
+		}
+	}
+
```

Comment:
And technically all this code should be in hiveadmission.go as that is specific to that component.

Otherwise looks good!